index.html

```diff
@@ -1618,52 +1649,323 @@ <h3>Methods inherited from {{ConsumedThing}}</h3>
     <section data-dfn-for="PropertyReadHandler">
       <h3>The <dfn>PropertyReadHandler</dfn> callback</h3>
       <p>
-        A function that is called when an external request for reading a <a>Property</a> is received and defines what to do with such requests. It returns a {{Promise}} and resolves it when the value of the <a>Property</a> matching the |name:string| argument is obtained, or rejects with an error if the property is not found or the value cannot be retrieved.
+        A function that is called when an external request for reading a <a>Property</a> is received and defines what to do with such requests. It returns a {{Promise}} and resolves with an {{InteractionData}} object
+        created by running the <a>create interaction data</a> steps on the
+        value of the <a>Property</a> matching the |name:string| argument that is obtained, or rejects with an error if the property is not found or the value cannot be retrieved.
+      </p>
```

Comment:
Actually I should not summarize an algorithm in that paragraph, just say what does it return.
It is the algorithmic steps that should specify what to do. Removing it.

Reply:
Ok, but I actually meant is that the promise should return `any`. So that the application does not need to worry about how to create an `InteractionData` in trivial cases. Am I missing something?

Reply:
We have agreed in the calls that reading functions always return `InteractionData` that has the `value()` convenience function and is created by the impl, while for writing functions we accept `any`, in order to relieve apps from creating `InteractionData` for trivial cases.

Reply:
That of course on ConsumedThing side.

On the ExposedThing side, the handlers also return `InteractionData`, and - now I get the question :) - you're asking why not allowing to return `any` in the handler functions and let the implementation bother with `InteractionData`. 

Well, a handler provides total control - an app may actually _want_ to _define_ how to create `InteractionData`. In fact you're right, we could apply the same principle we applied at ConsumedThing writes, i.e. allow returning `any`, and if that is an `InteractionData`, impls should just use it as it is, otherwise should use whatever is returned in order to construct a reply, possibly involving a Body-like object (which is InteractionData-like).

So all-in-all I think this is fair. I will modify the handler signatures.

Reply:
Done. Actually it's the exact opposite of how we do it in ConsumedThing.
When data comes from the implementation, it's `InteractionData`, when it goes there, it's `any`.

Reply:
I am not sure if I can agree.
` readProperty()` and `PropertyReadHandler` should return the same type. Either `any` or `InteractionData`.
Now we mix it again!?

Reply:
As discussed on the call on 15.6.2020, we leave the current interfaces and include an explanatory section on how `InteractionData` vs `any` are used.